math/genvector/inc/Math/GenVector/Transform3D.h

```diff
@@ -606,36 +868,96 @@ class Transform3D {
    /**
       make transformation from first a rotation then a translation
    */
-   void  AssignFrom( const Rotation3D & r, const Vector & v);
+   void AssignFrom( const Rotation3D & r, const Vector & v)
+   {
+     // assignment  from rotation + translation
+     
+     T rotData[9];
+     r.GetComponents(rotData, rotData +9);
+     // first raw
+     for (int i = 0; i < 3; ++i)
+       fM[i] = rotData[i];
+     // second raw
+     for (int i = 0; i < 3; ++i)
+       fM[kYX+i] = rotData[3+i];
+     // third raw
+     for (int i = 0; i < 3; ++i)
+       fM[kZX+i] = rotData[6+i];
+     
+     // translation data
+     T vecData[3];
+     v.GetCoordinates(vecData, vecData+3);
+     fM[kDX] = vecData[0];
+     fM[kDY] = vecData[1];
+     fM[kDZ] = vecData[2];
+   }
+
 
    /**
       make transformation from only rotations (zero translation)
    */
-   void  AssignFrom( const Rotation3D & r);
+   void  AssignFrom( const Rotation3D & r)
```

Comment:
I believe this is what you need to run:
```
git checkout -b cjones051073-GenVector-AddTemplates master
git pull https://github.com/cjones051073/root.git GenVector-AddTemplates
wget https://llvm.org/svn/llvm-project/cfe/trunk/tools/clang-format/git-clang-format
PATH=~/workspace/llvm-git/inst/bin/:$PATH ~/workspace/llvm-git/inst/bin/git-clang-format --commit 22a3099132 --diff
```
where PATH has to point to the clang-format binary.